storm-core/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -356,13 +358,14 @@ public static void sleep(long millis) {
         try {
             Time.sleep(millis);
         } catch(InterruptedException e) {
+            currentThread().interrupt();
```

Comment:
How it will collaborate with throwing RuntimeException? Are we assuming that caller catches RuntimeException? Just not clear that how it works together.